api/ondemand_vhds_rds_protocol.rst

```diff
@@ -0,0 +1,187 @@
+of maintaining route configs while VHDS is in charge of communicating virtual
+hosts. In VHDS, Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with type_url set to type.googleapis.com/envoy.api.v2.route.VirtualHost and
+resource_names set to a list of routeconfig names + the request HTTP "host"
+headers for which it would like configuration. The management server will
+respond with a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with the resources field populated with the specified virtual hosts, the name
+field populated with the virtual host name, and the alias field populated with
+the explicit (no wildcard) domains of the virtual host. Future updates to these
+virtual hosts will be sent via spontaneous updates.
+
+Unsubscribing from Virtual Hosts
+================================
+Virtual hosts can be unsubscribed to via a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with their route config names + domains provided in the
+resource_names_unsubscribe field. Envoy will remove any route config names +
+domains that it finds in the
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+removed_resources field.
+
+Subscribing to Virtual Hosts
+============================
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the resource_names_subscribe field populated with the route config names
++ domains of each of the resources that it would like to subscribe to. Each of
+the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+During spontaneous updates configuration server will only send updates for
+virtual hosts that Envoy is aware of -- the configuration server needs to
+keep track of virtual hosts known to Envoy.
+
+Updates of non-virtual host fields
+==================================
+It might be required to update non-virtual host fields of route configuration
+while keeping virtual hosts as is. This would probably happen during
+spontaneous updates only. In such a case a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+containing a route configuration with an empty virtual hosts list can be used.
+
+Compatibility with Scoped RDS
+=============================
+Both approaches appear to be compatible with scoped RDS proposal:
+route_configuration name can still be used for vhost matching, but with scoped
+RDS configured it would point to a scoped route configuration.
+
+The xDS management server will also support the ability of Envoy to tell it
```

Comment:
This appears to be unrelated to the SRDS section. I suggest adding a new section header to clearly delineate.

Reply:
Fixed. Thanks.